src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/pact/ReferralContracts.kt

```diff
@@ -0,0 +1,77 @@
+  fun `create the assigned referral`() {
+    setupAssistant.createAssignedReferral(id = UUID.fromString("2f4e91bf-5f73-4ca8-ad84-afee3f12ed8e"))
+  }
+
+  @State(
+    "There is an existing sent referral with ID of 400be4c6-1aa4-4f52-ae86-cbd5d23309bf",
+    "There are some existing sent referrals",
+  )
+  fun `create sent referral 400be4c6`() {
+    val referral = setupAssistant.createSentReferral(id = UUID.fromString("400be4c6-1aa4-4f52-ae86-cbd5d23309bf"))
+    setupAssistant.fillReferralFields(referral)
+    setupAssistant.mockServiceProviderOrganization(referral.intervention.dynamicFrameworkContract.serviceProvider.id)
+  }
+
+  @State("There is an existing draft referral with ID of ac386c25-52c8-41fa-9213-fcf42e24b0b5")
+  fun `create referral with the required id`() {
+    setupAssistant.createDraftReferral(id = UUID.fromString("ac386c25-52c8-41fa-9213-fcf42e24b0b5"))
+  }
+
+  @State(
+    "a draft referral with ID dfb64747-f658-40e0-a827-87b4b0bdcfed exists",
+    "a single referral for user with ID 8751622134 exists"
+  )
+  fun `create a new draft referral for 'deliusUser' with a specific createdAt timestamp and CRN`() {
+    setupAssistant.createDraftReferral(
+      id = UUID.fromString("dfb64747-f658-40e0-a827-87b4b0bdcfed"),
+      createdAt = OffsetDateTime.parse("2020-12-07T20:45:21.986389+00:00"),
+      serviceUserCRN = "X862134"
+    )
+  }
+
+  @State(
+    "There is an existing draft referral with ID of d496e4a7-7cc1-44ea-ba67-c295084f1962",
+    "There is an existing draft referral with ID of d496e4a7-7cc1-44ea-ba67-c295084f1962, and it has had a service category selected",
+    "There is an existing draft referral with ID of d496e4a7-7cc1-44ea-ba67-c295084f1962, and it has had a service provider selected",
+  )
+  fun `create a new draft referral with accommodation service category and 'HARMONY_LIVING' service provider set (both defaults)`() {
+    setupAssistant.createDraftReferral(id = UUID.fromString("d496e4a7-7cc1-44ea-ba67-c295084f1962"))
+  }
+
+  @State("There is an existing draft referral with ID of 037cc90b-beaa-4a32-9ab7-7f79136e1d27, and it has had desired outcomes selected")
+  fun `create a new draft referral and with desired outcomes`() {
+    val intervention = setupAssistant.createIntervention()
+    val desiredOutcomes = setupAssistant.desiredOutcomesForServiceCategory(intervention.dynamicFrameworkContract.serviceCategory.id)
+    val referral = setupAssistant.createDraftReferral(id = UUID.fromString("037cc90b-beaa-4a32-9ab7-7f79136e1d27"))
+    setupAssistant.fillReferralFields(referral, desiredOutcomes = desiredOutcomes)
+  }
+
+  @State("There is an existing draft referral with ID of 1219a064-709b-4b6c-a11e-10b8cb3966f6, and it has had a service user selected")
+  fun `create a new draft referral with the service user data expected`() {
+    val referral = setupAssistant.createDraftReferral(id = UUID.fromString("1219a064-709b-4b6c-a11e-10b8cb3966f6"))
+    setupAssistant.fillReferralFields(referral)
```

Comment:
Should we rethink how we describe our `@State`s?

I.e. it sounds like "an empty draft referral with ID ..." and "an existing ready-to-send draft referral with ID ..." encapsulates most of these states